ui/static/ui/js/listing.js

```diff
@@ -148,54 +175,121 @@ define('listing',
         Exports.loadExportsHeader(exportCount, $("#exports_heading")[0]);
       };
 
+      /**
+       * When called, query search API using query parameters of this URL
+       * and update listing with updated resources.
+       *
+       * @returns {jQuery.Deferred} A promise that's resolved or rejected when
+       * the AJAX call completes and the rerender is triggered.
+       */
+      var refreshFromAPI;
+
+      // Will be set in refreshFromAPI
+      var selectedFacets;
+
       var openResourcePanel = function(resourceId) {
         LearningResources.loader(
-          repoSlug, resourceId, $("#tab-1")[0]);
+          repoSlug, resourceId, refreshFromAPI, $("#tab-1")[0]);
         $('.cd-panel').addClass('is-visible');
         StaticAssets.loader(
           repoSlug, resourceId, $("#tab-3")[0]);
       };
 
-      var facetIsSelected = function(facetId, valueId) {
-        var facetQuery = "selected_facets=" + facetId + "_exact:" + valueId;
-        if (location.search === undefined) {
-          return false;
-        }
-        return location.search.indexOf(facetQuery) !== -1;
+      refreshFromAPI = function() {
+        var newQuery = "?" + URI().search(queryMap).query();
+        // Let's hope nobody's using IE 9 or earlier...
+        window.history.replaceState("currenturl", "MIT LORE", newQuery);
+
+        var url = "/api/v1/repositories/" +
+          listingOptions.repoSlug + "/search/" + newQuery;
+
+        $("#listing").css({opacity: 0.6});
+        return $.get(url).then(function(collection) {
+          listingOptions = $.extend({}, listingOptions);
+          listingOptions.resources = collection.results;
+          listingOptions.facetCounts = collection.facet_counts;
+          selectedFacets = collection.selected_facets;
+
+          numPages = Math.ceil(collection.count / 20);
```

Comment:
Why we just don't do something like
`numPages = Math.ceil(collection.count / collection.results.length);` ?
If we change the number of learning resources per page in the backend this will update automatically.